lib/web_ui/flutter_js/src/canvaskit_loader.js

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import { engineRevision } from "./revision.js";
+import { createWasmInstantiator } from "./instantiate_wasm.js";
+
+export const loadCanvasKit = (deps, config, browserEnvironment) => {
+  window.flutterCanvasKitLoaded = new Promise((resolve, reject) => {
+    const supportsChromiumCanvasKit = browserEnvironment.hasChromiumBreakIterators && browserEnvironment.hasImageCodecs;
+    if (!supportsChromiumCanvasKit && config.canvasKitVariant == "chromium") {
+      throw "Chromium CanvasKit variant specifically requested, but unsupported in this browser";
+    }
+    const useChromiumCanvasKit = supportsChromiumCanvasKit && (config.canvasKitVariant !== "full");
+    let baseUrl = config.canvasKitBaseUrl ?? `https://www.gstatic.com/flutter-canvaskit/${engineRevision}/`;
+    if (useChromiumCanvasKit) {
+      baseUrl = `${baseUrl}/chromium/`;
+    }
+    let canvasKitUrl = `${baseUrl}canvaskit.js`;
+    if (deps.flutterTT.policy) {
+      canvasKitUrl = deps.flutterTT.policy.createScriptURL(canvasKitUrl);
+    }
+    const wasmInstantiator = createWasmInstantiator(`${baseUrl}canvaskit.wasm`);
+    const script = document.createElement("script");
+    script.src = canvasKitUrl;
+    script.addEventListener('load', async () => {
+      try {
+        const canvasKit = await CanvasKitInit({
+          instantiateWasm: wasmInstantiator,
+        });
+        window.flutterCanvasKit = canvasKit;
+        resolve(canvasKit);  
+      } catch (e) {
+        reject(e);
+      }
+    });
+    script.addEventListener('error', (e) => {
+      reject(e);
+    });
+    document.head.appendChild(script);
+  });
+  return window.flutterCanvasKitLoaded;
```

Comment:
I'm not sure what you mean. It should never be null, it's a promise that is constructed immediately above. It is true that the promise isn't actually used by JavaScript (at least right now) and it's only used by the dart code later on initialization, but I figured returning the promise is probably a bit more flexible. I could be convinced otherwise though if you think returning it is a bad idea.

Reply:
Let me rephrase, because I can't even understand what I'm asking above:

Is it OK that if `loadCanvasKit` is called multiple times we create (and overwrite) the `window.flutterCanvasKitLoaded` promise multiple times, or should we always return the promise that was created the first time the method was called?

Reply:
Ah, I see. Good question. I think it would be weird (by which I mean wrong) for the user to call load multiple times. Maybe it would actually make sense to explicitly throw if they did that? Or, as you say, we could just return the promise and do nothing.